event/datadeal/data_verification_event.go

```diff
@@ -1 +1,215 @@
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/medibloc/panacea-core/v2/x/datadeal/types"
+	oracletypes "github.com/medibloc/panacea-core/v2/x/oracle/types"
+	"github.com/medibloc/panacea-doracle/config"
+	"github.com/medibloc/panacea-doracle/crypto"
+	"github.com/medibloc/panacea-doracle/event"
+	"github.com/medibloc/panacea-doracle/ipfs"
+	"github.com/medibloc/panacea-doracle/panacea"
+	"github.com/medibloc/panacea-doracle/sgx"
+	"github.com/medibloc/panacea-doracle/validation"
+	log "github.com/sirupsen/logrus"
+	ctypes "github.com/tendermint/tendermint/rpc/core/types"
+)
+
+var _ event.Event = (*DataVerificationEvent)(nil)
+
+type DataVerificationEvent struct {
+	reactor reactor
+}
+
+type reactor interface {
+	GRPCClient() *panacea.GrpcClient
+	EnclaveInfo() *sgx.EnclaveInfo
+	OracleAcc() *panacea.OracleAccount
+	OraclePrivKey() *btcec.PrivateKey
+	Config() *config.Config
+	QueryClient() *panacea.QueryClient
+	Ipfs() *ipfs.Ipfs
+}
+
+func NewDataVerificationEvent(r reactor) DataVerificationEvent {
+	return DataVerificationEvent{r}
+}
+
+func (d DataVerificationEvent) GetEventType() string {
+	return "message"
+}
+
+func (d DataVerificationEvent) GetEventAttributeKey() string {
+	return "action"
+}
+
+func (d DataVerificationEvent) GetEventAttributeValue() string {
+	return "'SellData'"
+}
+
+func (d DataVerificationEvent) EventHandler(event ctypes.ResultEvent) error {
+	dealIDStr := event.Events[types.EventTypeDataVerificationVote+"."+types.AttributeKeyDealID][0]
+	dataHash := event.Events[types.EventTypeDataVerificationVote+"."+types.AttributeKeyDataHash][0]
+
+	dealID, err := strconv.ParseUint(dealIDStr, 10, 64)
+	if err != nil {
+		return err
+	}
+
+	deal, err := d.reactor.QueryClient().GetDeal(dealID)
+	if err != nil {
+		return err
+	}
+
+	dataSale, err := d.reactor.QueryClient().GetDataSale(dataHash, dealID)
+	if err != nil {
+		return err
+	}
+
+	encryptedDataBz, err := d.reactor.Ipfs().Get(dataSale.VerifiableCid)
+	if err != nil {
+		return err
+	}
+
+	oraclePrivKey := d.reactor.OraclePrivKey()
+
+	decryptedData, err := d.decryptData(oraclePrivKey, dataSale, deal, encryptedDataBz)
+	if err != nil {
+		return err
+	}
+
+	voteOption := d.verifyDataSaleAndGetVoteOption(dataSale, decryptedData, deal.DataSchema)
+
+	msgVoteDataVerification, err := makeDataVerificationVote(
+		d.reactor.OracleAcc().GetAddress(),
+		dataHash,
+		dealID,
+		voteOption,
+		oraclePrivKey.Serialize(),
+	)
+	if err != nil {
+		return err
+	}
+
+	txBuilder := panacea.NewTxBuilder(*d.reactor.QueryClient())
+
+	txBytes, err := generateTxBytes(msgVoteDataVerification, d.reactor.OracleAcc().GetPrivKey(), d.reactor.Config(), txBuilder)
+	if err != nil {
+		return err
+	}
+
+	if err := broadcastTx(d.reactor.GRPCClient(), txBytes); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (d DataVerificationEvent) decryptData(oraclePrivKey *btcec.PrivateKey, dataSale *types.DataSale, deal *types.Deal, encryptedDataBz []byte) ([]byte, error) {
```

Comment:
Currently, this function do two things: getting account and decrypting data.
I suggest to limit this function to do decrypt data because its name is `decryptData`.